typescript.md

```diff
@@ -0,0 +1,73 @@
+TypeScript Code Style
+=====================
+
+## Table of Contents
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+- [General](#general)
+- [Naming](#naming)
+- [Typing](#typing)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## General
+
+This style guide is generally the same as the one for JavaScript.
+
+[&#8593; back to TOC](#table-of-contents)
+
+## Naming
+
+* `IInterfaceLikeThis` — notice `I` in the beginning.
+* In generics:
+  * `K` is reserved for key-like data types;
+  * `T` and `U` are reserved for generic data types;
+  * other generic parameters should have meaningfull names.
```

Comment:
Sometimes in external declarations I see `T` prefix for type variables:

```ts
Promise<TAll>(...)
```
Personally, I don't like it. Maybe its should be forbidden.